dropwizard-demo/test/com/test/ControllerTest.java

```diff
@@ -0,0 +1,132 @@
+
+
+    @Before
+    public void setUp() {
+        brsConfiguration = Mockito.mock(BRSConfiguration.class);
+        tweetPost = Mockito.mock(Controller.class);
+    }
+
+    @Test
+    public void testcase1_getTweets() {
+        when(brsConfiguration.configurationBuilder()).thenReturn(new ConfigurationBuilder());
+        ArrayList<String> str = new ArrayList<String>();
+        str.add("hlo");
+        when(tweetPost.GetTweets()).thenReturn(Response.ok(str).build());
+        ArrayList<String> arrayList = new ArrayList<String>();
+        arrayList.add("hlo");
+        Response expectedTweet = Response.ok(arrayList).build();
+        Response actualTweet = tweetPost.GetTweets();
+        Assert.assertEquals(expectedTweet.getStatus(), actualTweet.getStatus());
+        Assert.assertEquals(expectedTweet.getStatus(),actualTweet.getStatus());
+    }
+
+    @Test
+    public void testcase_noTweetsFound() throws TwitterException {
+        when(brsConfiguration.configurationBuilder()).thenReturn(new ConfigurationBuilder());
+        ArrayList<String> expectedTweet = null;
+        when(tweetPost.GetTweets()).thenReturn(Response.ok().build());
+        ArrayList<String> actualTweet = Controller.getTweets();
+        if (actualTweet.isEmpty()) {
+            actualTweet = null;
+        }
+        Assert.assertEquals(expectedTweet, actualTweet);
+    }
+
+    @Test
+    public void testcase2_getTweets() throws TwitterException {
+        when(brsConfiguration.configurationBuilder()).thenReturn(new ConfigurationBuilder());
+        Twitter twitter = TwitterFactory.getSingleton();
+        ArrayList<String> arrayList = new ArrayList<String>();
+        List<Status> status = twitter.getHomeTimeline();
+        for (Status st : status) {
+            arrayList.add(st.getText());
+        }
+        ArrayList<String> expectedTweet = arrayList;
+        ArrayList<String> actualTweet = Controller.getTweets();
+        Assert.assertEquals(expectedTweet, actualTweet);
+    }
+
+    @Test
+    public void testcase_sendTweet_success() throws TwitterException {
+        when(brsConfiguration.configurationBuilder()).thenReturn(new ConfigurationBuilder());
+        Twitter twitter = TwitterFactory.getSingleton();
+        String message = "Testing Twitter4J, posting to Twitter programatically hi .";
+        Status status = twitter.updateStatus(message);
+        System.out.println("Successfully updated the status to [" + status.getText() + "].");
+    }
+
+    @Test
+    public void testSearch_finder() throws TwitterException {
+        when(brsConfiguration.configurationBuilder()).thenReturn(new ConfigurationBuilder());
+        Twitter twitter = TwitterFactory.getSingleton();
+        Query query = new Query("source:twitter4j yusukey");
+        QueryResult result = twitter.search(query);
+        for (Status status : result.getTweets()) {
+          System.out.println("@" + status.getUser().getScreenName() + ":" + status.getText());
+        }
```

Comment:
Assert missing, what are u testing here, if there is no assert
what is the use of test case if it never checks/asserts anything?

Reply:
I am still working on it.